torchaudio/models/decoder/_ctc_decoder.py

```diff
@@ -218,7 +219,11 @@ def ctc_decoder(
             decoding.
         tokens (str or List[str]): file or list containing valid tokens. If using a file, the expected
             format is for tokens mapping to the same index to be on the same line
-        lm (str or None, optional): file containing language model, or `None` if not using a language model
+        lm (str, CTCDecoderLM, or None, optional): either a path containing KenLM language model,
```

Comment:
would this need to be moved to the end for bc-breaking purposes?

Reply:
I think it's okay to introduce BC-breaking change. having two related arguments 10 arguments away seems not a great UX, and I bet many people use keyword argument here.

Reply:
should we add a warning inside the function to indicate this argument change?

Reply:
I think we can do type check on `lm_dict` value.
We want to avoid the situation where things keep working but in a wrong way. (though I doubt that happens)

Users using keywords arguments are not affected, but those pass `nbest` as positional argument are affected.
So ensuring that `lm_dict` is `None` or `str` should suffice.